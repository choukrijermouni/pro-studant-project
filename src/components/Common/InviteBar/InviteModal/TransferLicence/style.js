import {
  neutral,
  blue,
  red
} from '@pro_boa/ui'

export default {
  modal: {
    width: 837,
    backgroundColor: neutral[0],
    borderRadius: 4,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  header: {
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'column',
    alignItems: 'center'
  },
  controlStyle: {
    width: '100%'
  },
  textStyle: {
    display: 'flex',
    alignItems: 'center'
  },
  invitesBox: {
    height: 72,
    width: 548,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center ',
    alignItems: 'center',
    borderRadius: 4,
    backgroundColor: neutral[0],
    boxShadow: '0 0 36px 0 rgba(194,212,230,0.23)'
  },
  invites: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center ',
    alignItems: 'center'
  },
  icon: {
    fontSize: 24,
    color: blue[1]
  },
  editIcon: {
    fontSize: 15,
    color: neutral[3],
    padding: '11px 12px',
    border: `1px solid ${neutral[3]}`,
    borderRadius: '50%',
    marginLeft: 14
  },
  licenseButtons: {
    display: 'flex',
    justifyContent: 'space-between'
  },
  line: {
    backgroundColor: neutral[13],
    height: 1,
    border: 'none',
    marginBottom: 20,
    marginTop: 45
  },
  circle: {
    borderRadius: '50%',
    width: 15,
    height: 15,
    padding: 2,
    background: red[2],
    color: neutral[0],
    textAlign: 'center',
    font: '13px Arial, sans-serif',
    left: -10,
    top: -3,
    position: 'relative'
  }
}
