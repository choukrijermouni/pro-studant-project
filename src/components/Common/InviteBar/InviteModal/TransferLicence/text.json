{
  "inviteTitle": "Inviter des apprenants",
  "subHeader": {
    "first": "Vous avez invité ",
    "second": "nouveaux apprenants à rejoindre votre organisme"
  },
  "transfer": "Confirmer le transfert de la licence",
  "subTitle": "Une fois que la licence est transférée d'un apprenant à l'autre : ",
  "dot": "•",
  "steps": [
    "Le nouveau bénéficiaire reçoit un e-mail de transfert de licence",
    "L’ancien bénéficiaire n'aura plus de licence",
    "Le nouveau bénéficiaire reçoit un abonnement mensuel d'une durée de 2 mois et 4 jours."
  ],
  "confirmButton": "Confirmer",
  "abortButton": "Annuler",
  "confirm": "Êtes-vous sûr de vouloir transférer la licence ?",
  "support": "En cas de besoin, contactez-nous par e-mail à l'adresse :",
  "orViaPhone": "ou par téléphone au :"
}