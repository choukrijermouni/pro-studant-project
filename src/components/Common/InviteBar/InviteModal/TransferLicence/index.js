import {
  TypographyElement,
  Button,
  ButtonVariation,
  ButtonSize,
  blue,
  neutral
} from '@pro_boa/ui'
import { createUseStyles } from 'react-jss'
import style from './style'
import Text from './text.json'
import { transferLicence } from 'assets'
import { salesB2B } from 'constants/'

const useStyles = createUseStyles(style)
export default ({ nextStep, closeModal }) => {
  const { modal, header, controlStyle, textStyle, licenseButtons, line } = useStyles()
  return (
    <div>
      <div className={modal}>
        <div className={header}>
          <img src={transferLicence} alt='invitation' />
          <TypographyElement
            component='h2'
            variant='heading2'
            fontSize='29px'
            align='left'
            spacing='20px 0 8px 0'
          >
            {Text.transfer}
          </TypographyElement>
          <div>
            <TypographyElement
              component='h2'
              variant='heading2'
              align='left'
              fontSize='16px'
              spacing='16px 0 0 0'
            >
              {Text.subTitle}
            </TypographyElement>
            {Text.steps.map((step, key) =>
              <div className={textStyle} key={key}>
                <TypographyElement
                  component='h2'
                  variant='heading2'
                  fontSize='24px'
                  align='left'
                  display='inline'
                  lineHeight='0px'
                  spacing='10px 8px 0 0'
                  color={blue[0]}
                >
                  {Text.dot}
                </TypographyElement>
                <TypographyElement
                  component='h2'
                  variant='heading2'
                  fontSize='14px'
                  align='left'
                  display='inline'
                  spacing='10px 0 0 0'
                  lineHeight='25px'
                  color={neutral[4]}
                >
                  {step}
                </TypographyElement>
              </div>
            )}
            <TypographyElement
              component='h2'
              variant='heading2'
              fontSize='14px'
              align='left'
              lineHeight='25px'
              spacing='16px 0 5px 0'
              color={neutral[5]}
            >
              {Text.confirm}
            </TypographyElement>
            <TypographyElement
              component='h2'
              variant='heading2'
              fontSize='14px'
              align='left'
              lineHeight='25px'
              color={neutral[4]}
            >
              {`${Text.contactText} ${salesB2B.SalesEmail} ${Text.orViaPhone} ${salesB2B.SalesPhone}`}
            </TypographyElement>
          </div>
        </div>
        <div className={controlStyle}>
          <hr className={line} />
          <div className={licenseButtons}>
            <Button variation={ButtonVariation.secondary} size={ButtonSize.big} width={153} height={54} label={Text.abortButton} handleClick={closeModal} />
            <Button variation={ButtonVariation.primary} size={ButtonSize.big} width={153} height={54} label={Text.confirmButton} handleClick={nextStep} />
          </div>
        </div>
      </div>
    </div>
  )
}
