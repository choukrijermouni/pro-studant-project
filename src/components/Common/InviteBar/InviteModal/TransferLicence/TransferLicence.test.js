import { shallow, configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import renderer from 'react-test-renderer'
import TransferLicence from '.'
configure({ adapter: new Adapter() })

describe('>>> TransferLicence --- Shallow Render TransferLicence component', () => {
  let wrapper
  let tree

  beforeEach(() => {
    wrapper = shallow(<TransferLicence />)
    tree = renderer.create(
      <TransferLicence />
    ).toJSON()
  })

  it('+++ render correctly and match snapshot', () => {
    expect(tree).toMatchSnapshot()
  })

  it('+++ render the component', () => {
    expect(wrapper).toBeTruthy()
  })
})
