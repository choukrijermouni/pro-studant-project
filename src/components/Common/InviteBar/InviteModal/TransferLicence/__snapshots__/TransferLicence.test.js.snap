// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> TransferLicence --- Shallow Render TransferLicence component +++ render correctly and match snapshot 1`] = `
<div>
  <div
    className="modal-0-2-1"
  >
    <div
      className="header-0-2-2"
    >
      <img
        alt="invitation"
        src={Object {}}
      />
      <p
        className="text-0-2-12 text-d0-0-2-13"
      >
        Confirmation du transfert de la licence
      </p>
      <div>
        <p
          className="text-0-2-14 text-d0-0-2-15"
        >
          Après la confirmation du transfert de la licence voilà ce qui vas se passer :
        </p>
        <div
          className="textStyle-0-2-4"
        >
          <p
            className="text-0-2-16 text-d0-0-2-17"
          >
            •
          </p>
          <p
            className="text-0-2-18 text-d0-0-2-19"
          >
            Nous enverrons un e-mail de transfert/ à : p.germain@tesla.com
          </p>
        </div>
        <div
          className="textStyle-0-2-4"
        >
          <p
            className="text-0-2-20 text-d0-0-2-21"
          >
            •
          </p>
          <p
            className="text-0-2-22 text-d0-0-2-23"
          >
            L’apprenant (p.germain@tesla.com) sera sans licence
          </p>
        </div>
        <div
          className="textStyle-0-2-4"
        >
          <p
            className="text-0-2-24 text-d0-0-2-25"
          >
            •
          </p>
          <p
            className="text-0-2-26 text-d0-0-2-27"
          >
            L'apprenant (j.mark@tesla.com) va reçu un abonnement Mensuel avec la durée de 2 mois et 4 jours.
          </p>
        </div>
        <p
          className="text-0-2-28 text-d0-0-2-29"
        >
          Voulez-vous vraiment transférer la licence ?
        </p>
        <p
          className="text-0-2-30 text-d0-0-2-31"
        >
          Si vous avez des questions, contactez-nous via info@pro_boa.com ou appelez-nous au  +33 01 77 62 45 80
        </p>
      </div>
    </div>
    <div
      className="controlStyle-0-2-3"
    >
      <hr
        className="line-0-2-10"
      />
      <div
        className="licenseButtons-0-2-9"
      >
        <button
          className="button-0-2-32 button-d0-0-2-34"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-36 text-d0-0-2-37 text-0-2-33 text-d1-0-2-35"
          >
            Annuler
          </p>
        </button>
        <button
          className="button-0-2-38 button-d0-0-2-40"
          onClick={[Function]}
          type="button"
        >
          <p
            className="text-0-2-42 text-d0-0-2-43 text-0-2-39 text-d1-0-2-41"
          >
            Confirmer
          </p>
        </button>
      </div>
    </div>
  </div>
</div>
`;
